### ISSUE TITLE
Commit to a single code style in the code base and mandate it with automated tool

### ISSUE THREAD

**Issue by DolceTriade** [Unvanquished/Unvanquished#2207]
We have a lot of code styles in our code base. Let's decide on a single style we. Subsequent commits will be required to be formatted using clang-format.

I propose this code style: https://gist.github.com/DolceTriade/94250674f24bd4465c6d00ff2e0c99aa, but that's just because that's the legacy q3 style. If people prefer other styles, then please provide an appropriate clang-format file. I don't especially care which style we use as long as we decide on one and we have appropriate tooling to mandate it.

We can use a file versioned in the repo to ignore the bulk formatting commits so we don't lose history: https://www.moxio.com/blog/43/ignoring-bulk-change-commits-with-git-blame

I see two approaches here:

 1. Do it gradually. Any code that is touched can be formatted with the new style. If it touches a function using a different style, the whole function should be reformatted (or the entire logical code block if it is a class or something...).
 2. Do a one time reformat of the code base.  This will need to be done with care as any outstanding PRs will conflict. Therefore we will need to have to merge all outstanding PRs and ensure all forks of the project are aware of our plans.

Let's decide on this and move forward. Having many different styles in the engine sucks and it sucks that we spend a shit load of time in PR reviews bikeshedding on code style issues. Doing this once and mandating this style with a CI check will basically eliminate all of these discussions and allow us to focus on the design and concepts of the code.

**Comment by slipher** [comment]
https://wiki.unvanquished.net/wiki/Reformatting_the_source some notes on the subject

**Comment by DolceTriade** [comment]
Yes, but the wiki is not a mandate. Hopefully we can come to a decision here and actually enforce it.

**Comment by DolceTriade** [comment]
I propose we move forward with @slipher's suggestion.

let's make sure any large outstanding PRs are merged and then let's just do this.

I propose we use this style: https://gist.githubusercontent.com/DolceTriade/32b1c00498035f96afc848524f5ec459/raw/aea34e78ed9bd4599723fac6538014ea0f07e748/.clang-format

Open to other suggestions.
